### PR TITLE
Add tests for runtime exception if device doesn't support required work group size 

### PR DESCRIPTION
**Pull request by kopylovanat**
This PR depends on https://github.com/KhronosGroup/SYCL-CTS/pull/287
Changes related to this PR in the file `kernel_features_reqd_work_group_size.cpp`

### CODE REVIEW

File: tests/optional_kernel_features/kernel_features_reqd_work_group_size.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  auto max_wg_size =
+      queue.get_device().get_info<sycl::info::device::max_work_group_size>();
+
+  bool is_exception_expected = (N > max_wg_size);
+
+  // Set expected error code
+  constexpr sycl::errc expected_errc = sycl::errc::kernel_not_supported;
+
+  {
+    const auto lambda_no_arg = []() [[sycl::reqd_work_group_size(N)]]{};
```
Comment (keryell) [inline tests/optional_kernel_features/kernel_features_reqd_work_group_size.cpp:39]:
```suggestion
    const auto lambda_no_arg = [] [[sycl::reqd_work_group_size(N)]]{};
```

**Review by keryell — APPROVED** [review]
Thanks.

File: tests/optional_kernel_features/kernel_features_reqd_work_group_size.cpp (hunk 1 above)
Comment (kopylovanat) [reply tests/optional_kernel_features/kernel_features_reqd_work_group_size.cpp]: Done.

**Comment by kopylovanat** [comment]
CI failed on ComputeCpp with [error](https://github.com/KhronosGroup/SYCL-CTS/runs/5355967729?check_suite_focus=true#step:8:2853) It's doesn't support implementation-defined kernel names according to paragraph [4.9.4.2](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#subsec:invokingkernels).

**Comment by keryell** [comment]
It looks like there are still room to code factorization along rebasing on https://github.com/KhronosGroup/SYCL-CTS/pull/287

**Comment by kopylovanat** [comment]
> It looks like there are still room to code factorization along rebasing on #287

Merge changes from https://github.com/KhronosGroup/SYCL-CTS/pull/287. Do you have some additional factorization in mind for this test?

**Pull request merged**